newIDE/app/public/external/yarn/yarn-main.js

```diff
@@ -0,0 +1,77 @@
+import { createPathEditorHeader, fileExists } from '../utils/path-editor.js';
+
+const electron = require('electron');
+const electronWindow = electron.remote.getCurrentWindow();
+const ipcRenderer = electron.ipcRenderer;
+const fs = require('fs');
+const remote = electron.remote;
+
+let yarn = null;
+let receivedData;
+
+const saveAndClose = pathEditor => {
+  const savePath = pathEditor.state.fullPath;
+  yarn.data.saveTo(savePath, yarn.data.getSaveData('json'), () => {
+    ipcRenderer.send('yarn-changes-saved', savePath, receivedData);
+    remote.getCurrentWindow().close();
+  });
+};
+
+const closeWindow = () => {
+  remote.getCurrentWindow().close();
+};
+
+const editorFrameEl = document.getElementById('yarn-frame');
+window.addEventListener('yarnReady', e => {
+  yarn = e;
+  yarn.app.fs = fs;
+  ipcRenderer.send('yarn-ready');
+});
+editorFrameEl.src = 'yarn-editor/app/index.html';
+
+// Called to load yarn data. Should be called after the window is fully loaded.
+
+ipcRenderer.on('yarn-open', (event, receivedOptions) => {
+  if (!yarn) return;
+
+  //Make a header
+  const pathEditorHeaderDiv = document.getElementById('path-editor-header');
+  const pathControl = createPathEditorHeader({
```

Comment:
Nitpicking: the function is called "createPathEditorHeader" so can we call the variable "pathEditorHeader"?